src/java/org/apache/cassandra/index/sai/disk/vector/CassandraOnHeapGraph.java

```diff
@@ -356,9 +361,29 @@ public SegmentMetadata.ComponentMetadataMap writeData(IndexDescriptor indexDescr
                                                         ? x -> ordinalMap.inverse().getOrDefault(x, x)
                                                         : x -> x;
 
+            // Retrieve the first compressed vectors with more than MAX_PQ_TRAINING_SET_SIZE rows
+            var indexes = new ArrayList<>(indexContext.getView().getIndexes());
+            indexes.sort(Comparator.comparing(ssTableIndex -> ssTableIndex.getSSTable().descriptor.id));
```

Comment:
is this newest-first or oldest-first?

Reply:
Good catch. I switched this to use `CompactionSSTable.maxTimestampDescending` since that avoids some of the type parameterization issues related to comparing sstable ids.